core/src/main/java/io/fabric8/maven/core/util/KubernetesResourceUtil.java

```diff
@@ -402,7 +404,19 @@ public static String extractContainerName(MavenProject project, ImageConfigurati
     private static String extractImageUser(String image, MavenProject project) {
         ImageName name = new ImageName(image);
         String imageUser = name.getUser();
-        return imageUser != null ? imageUser : project.getGroupId();
+        String projectGroupId = project.getGroupId();
+        if(imageUser != null) {
+            return imageUser;
+        } else {
+            if(projectGroupId == null || projectGroupId.matches(CONTAINER_NAME_REGEX)) {
+                return projectGroupId;
+            }
+            else {
+                return projectGroupId.replaceAll("[^a-zA-Z0-9-]", "")
+                        .replaceFirst("^-", "\\1")
```

Comment:
Are you sure that this works ? You have no capturing group, so what should `\1` refer to ? Actually you could use `.replaceAll("^-*","")` if you want to do it two steps.

Reply:
I tested it on groupIds like `io.fabric8-test-`, `io.fabric8-test-` and `-io.fabric8-test-`, Sorry I'm not an expert of regexes so might be doing some  mistake. We can replace `-` with any character in the desired container characters range, I just thought replacing it with `1`. Since you also suggested this in previous comment.

I tried your regex also i.e `projectGroupId.replaceAll("[^a-zA-Z0-9-]", "").replaceFirst("^-*(.*)-*$","\\1")` but it seems to be replacing everything with 1

Reply:
No, that `\1` references the content capturing group in the previous expression `(...)`. Its not "1" literally ;). Let me suggest a working solution and push it to your branch ...

Reply:
If you are interested how this works, look for "regular expression capturing group"